tests/unit/package_managers/python/uv/test_models.py

```
@@ -0,0 +1,739 @@
+        name="example", version="1.0.0", source=source, sdist=sdist, wheels=wheels or []
+    )
+
+
+def validate_source(raw: dict[str, Any]) -> PackageSource:
+    """Validate a raw uv.lock source table the only way production does, through UvPackage."""
+    return UvPackage.model_validate({"name": "example", "source": raw}).source
+
+
+class TestPackageSourceNormalization:
+    @pytest.mark.parametrize(
+        "raw, expected_model, expected_kind, expected_location",
```

Ruff (0.16.2)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)